src/main/java/life/qbic/portal/portlet/OMEROClientPortlet.java

```diff
@@ -486,14 +472,14 @@ private JsonStructure post(String url, Map<String, String> params)
     public Map<String, String> omeroJsonGetURLs() throws Exception {
         JsonObject json = (JsonObject) get(requestURL, null);
 
-        this.serviceURLs = new HashMap<String, String>();
+        Map<String, String> urls = new HashMap<>();
```

Comment:
I think its nice to have the type explicitly stated, but why not HashMap?

Reply:
Because we do not need it. For us it does not matter what type of Map the URLs are but only that they are provided by something implementing the Map interface.

Reply:
>The use of implementation types as object references limits your ability to use alternate implementations in the future as requirements change. Whenever available, referencing objects by their interface types provides much more flexibility.

http://www.ing.iac.es/~docs/external/java/pmd/rules/coupling.html#loosecoupling